tests/lit-tests/bool-store-varying.ispc

```diff
@@ -0,0 +1,25 @@
+// RUN: %{ispc} --target=host --nowrap --nostdlib -O2 --emit-llvm-text %s -o - | FileCheck %s
+
+// CHECK-LABEL: @y
+// CHECK: local_unnamed_addr global <[[WIDTH:.*]] x i8> zeroinitializer
+
+bool y;
+
+// CHECK-LABEL: @foo___vyTvyT(
+// CHECK-NEXT: allocas:
+// CHECK-NEXT: [[CMP:%.*]] = icmp ult <[[WIDTH]] x i8> %a, %b
+// CHECK-NEXT: [[CAST:%.*]] = sext <[[WIDTH]] x i1> [[CMP]] to <[[WIDTH]] x i8>
+// CHECK-NEXT: store <[[WIDTH]] x i8> [[CAST]], {{.*}} @y
+// CHECK-NEXT: ret void
+
+void foo(uint8 a, uint8 b)
+{
+    y = a < b;
+}
+
+// Note:
+// It is quite suspicous that bool is stored in memory as <WIDTH x i8> but
```

Comment:
How it's stored is governed by compatibility with C++. How it's passed is governed by platform ABI.

And yes, `bool` is a unique type in the sense that it's the only type that has different storage representation and register representation.

Reply:
I would suggest taking out this comment and explaining why it's different.

Reply:
I've changed it.

Reply:
One more thing I have noticed, that on `avx512-x32` target `uniform uint8<4>` vector has `<8 x i8>` type and `avx512-x64` has `<16 x i8>`. It looks intentional from code here: https://github.com/ispc/ispc/blob/5725fb8d495980af52b417ae0c7a89e2a258d2bd/src/type.cpp#L1823

but it is not clear for me why.

Reply:
The logic seems good, but I have a question why the next power of two is computed differently, while it should be the same.

Reply:
`nextPow2` is 4, but for all `g->target->getDataTypeWidth()` less than 32 (e.g., `avx512skx-x32` - 16, `avx512skx-x32` - 8) this returns not nextPow2 but `128 / g->target->getDataTypeWidth()`, i.e., 128/16 = 8 and 128/8 = 16 correspondingly